components/ReportSwitch.vue

```diff
@@ -18,17 +18,21 @@ const props = defineProps({
     required: true,
   },
 });
-const { toggleEulaPackageApproval } = useToggleApproval({
-  pkg: props.name,
-  eula: props.criterion,
-});
+const { toggleEulaPackageApproval } = useToggleApproval();
 const { report } = useSingleReport();
 const currentPackage = computed(() => report.value.state?.[props.name] || {});
 const checked = ref(currentPackage.value[props.criterion]);
 
 watchEffect(() => {
   checked.value = currentPackage.value[props.criterion];
 });
+
+const onToggle = (value) =>
+  toggleEulaPackageApproval({
+    value,
+    pkg: props.name,
+    eula: props.criterion,
```

Comment:
**question:** It seems like `toggleEulaPackageApproval` has been changed to take no arguments, so I am not sure how these are being passed through?

Reply:
The hook itself has been changed to take no arguments what this does is pass the arguments to the onMutate function so that we make sure there are fetched again the moment you click the button to make the changes